packages/oscal-react-library/src/components/OSCALProperties.tsx

```diff
@@ -0,0 +1,179 @@
+        {namespace}
+      </OSCALSystemImplementationTableTitle>
+      <TableContainer sx={{ maxHeight: "20em" }}>
+        <Table aria-label="Components" sx={{ height: "max-content" }}>
+          <StyledTableHead>
+            <TableRow>
+              <StyledHeaderTableCell>Name</StyledHeaderTableCell>
+              <StyledHeaderTableCell>Class</StyledHeaderTableCell>
+              <StyledHeaderTableCell>Value</StyledHeaderTableCell>
+            </TableRow>
+          </StyledTableHead>
+          <TableBody>
+            {properties
+              ?.filter((property) => namespaceOf(property?.ns) === namespace)
+              .map((property) => (
+                <OSCALProperty property={property} key={`$namespaceOf(property?.ns)-properties`} />
+              ))}
+          </TableBody>
+        </Table>
+      </TableContainer>
+    </>
+  );
+};
+
+interface OSCALPropertiesDialogProps {
+  /**
+   * Contains properties elements
+   */
+  properties?: Property[];
+  /**
+   * Title of the element the properties resides under
+   */
+  title?: string | ReactElement;
+}
+
+export const OSCALPropertiesDialog: React.FC<OSCALPropertiesDialogProps> = ({
+  properties,
+  title,
+}) => {
+  const [open, setOpen] = React.useState(false);
+
+  if (!properties) {
+    return null;
+  }
+  const namespaceToProps: Record<string, Property[]> = {};
+  for (const prop of properties) {
+    namespaceToProps[namespaceOf(prop.ns)] ??= [];
+    namespaceToProps[namespaceOf(prop.ns)].push(prop);
+  }
+  const { [NIST_DEFAULT_NAMESPACE]: nist, ...thirdParties } = namespaceToProps;
+
+  const handleOpen = () => {
+    setOpen(true);
+  };
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <>
+      <OSCALPropertiesButton
+        size="small"
+        variant="outlined"
+        onClick={handleOpen}
+        aria-label={`${title} properties button`}
+        startIcon={<HomeIcon />}
+      >
+        Properties
+      </OSCALPropertiesButton>
+      <Dialog
+        open={open}
+        onClose={handleClose}
+        scroll="paper"
+        aria-labelledby="scroll-dialog-title"
+        aria-describedby="scroll-dialog-description"
+        maxWidth="md"
+        fullWidth
+        sx={{ maxHeight: "75em" }}
+      >
+        <DialogContent sx={{ maxHeight: "75vh" }} dividers>
+          <DialogTitle id="scroll-dialog-title">{title} Properties</DialogTitle>
+          {/* Handle NIST properties */}
+          <OSCALProperties
+            properties={nist?.sort(byName)}
+            namespace={NIST_DEFAULT_NAMESPACE}
+            key={NIST_DEFAULT_NAMESPACE}
+          />
+          {
+            /* Handle 3rd party properties */
+            Object.entries(thirdParties)
+              .sort(([key1], [key2]) => key1.localeCompare(key2))
+              .map(([key, props]) => (
+                <OSCALProperties properties={props?.sort(byName)} namespace={key} key={key} />
```

Comment:
```suggestion
                <OSCALProperties properties={props.sort(byName)} namespace={key} key={key} />
```

Reply:
Several of the tests will fair when I remove either this optional chaining operator or the one in `nist?.sort(byName)`

Reply:
I'm wrong. It is possible to have properties defined and only NIST properties, or have properties defined and only 3rd party properties. `nist?.sort(byName)` must have the optional chaining operator, but it should be fine to remove it from the place you mentioned.  With the 3rd party properties filter, it will handle if there are no 3rd parties defined, so I should be able to remove the optional chain operator here: `props?.sort(byName)`. Although the type declaration should also account for there being no NIST properties present - which is a case where there would be passed no properties passed to `OSCALProperties`.